sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -507,60 +538,47 @@ def post_port_dom_info_to_db(logical_port_name, port_mapping, table, stop_event=
-                         ('tx1power', dom_info_dict['tx1power']),
-                         ('tx2power', dom_info_dict['tx2power']),
-                         ('tx3power', dom_info_dict['tx3power']),
-                         ('tx4power', dom_info_dict['tx4power']),
-                         ('tx5power', dom_info_dict['tx5power']),
-                         ('tx6power', dom_info_dict['tx6power']),
-                         ('tx7power', dom_info_dict['tx7power']),
-                         ('tx8power', dom_info_dict['tx8power'])
-                         ])
-                else:
-                    fvs = swsscommon.FieldValuePairs(
-                        [('temperature', dom_info_dict['temperature']),
-                         ('voltage', dom_info_dict['voltage']),
-                         ('rx1power', dom_info_dict['rx1power']),
-                         ('rx2power', dom_info_dict['rx2power']),
-                         ('rx3power', dom_info_dict['rx3power']),
-                         ('rx4power', dom_info_dict['rx4power']),
-                         ('tx1bias', dom_info_dict['tx1bias']),
-                         ('tx2bias', dom_info_dict['tx2bias']),
-                         ('tx3bias', dom_info_dict['tx3bias']),
-                         ('tx4bias', dom_info_dict['tx4bias']),
-                         ('tx1power', dom_info_dict['tx1power']),
-                         ('tx2power', dom_info_dict['tx2power']),
-                         ('tx3power', dom_info_dict['tx3power']),
-                         ('tx4power', dom_info_dict['tx4power'])
-                         ])
+                fvs = swsscommon.FieldValuePairs([(k, v) for k, v in dom_info_dict.items()])
+                table.set(physical_port_name, fvs)
+            else:
+                return SFP_EEPROM_NOT_READY
 
-                table.set(port_name, fvs)
+        except NotImplementedError:
+            helper_logger.log_error("This functionality is currently not implemented for this platform")
+            sys.exit(NOT_IMPLEMENTED_ERROR)
+
+# Update port pm info in db
+
+
+def post_port_pm_info_to_db(logical_port_name, port_mapping, table, stop_event=threading.Event(), pm_info_cache=None):
+    for physical_port, physical_port_name in get_physical_port_name_dict(logical_port_name, port_mapping).items():
+        if stop_event.is_set():
+            break
+
+        if not _wrapper_get_presence(physical_port):
+            continue
 
+        try:
+            if pm_info_cache is not None and physical_port in pm_info_cache:
+                # If cache is enabled and pm info is in cache, just read from cache, no need read from EEPROM
+                pm_info_dict = pm_info_cache[physical_port]
+            else:
+                pm_info_dict = _wrapper_get_transceiver_pm(physical_port)
+                if pm_info_cache is not None:
+                    # If cache is enabled, put dom information to cache
+                    pm_info_cache[physical_port] = pm_info_dict
```

Comment:
where are we going to use this pm_info_cache[physica_port] ? I dont see it being used

Reply:
pm_info_cache is passed to post_port_pm_info_to_db() from task_worker() of ```class DomInfoUpdateTask``` (in task_worker, we already have similar cache logic for dom_info dom_th_info in today's code)
pm_info_cache is used here:
```  
        if pm_info_cache is not None and physical_port in pm_info_cache:
            # If cache is enabled and pm info is in cache, just read from cache, no need read from EEPROM
            pm_info_dict = pm_info_cache[physical_port]

``` 
One possible situation I can think of: task_worker() later walks through a different logical port which belongs to a previously visited physical port, then cache is hit.